src/com/google/enterprise/adaptor/prebuilt/RegexFilter.java

```diff
@@ -0,0 +1,299 @@
+    }
+
+    @Override
+    public String toString() {
+      return name;
+    }
+  };
+
+  private static final Logger log
+      = Logger.getLogger(RegexFilter.class.getName());
+
+  /** The name of the key (either Metadata key or params key) to match. */
+  private String key;
+
+  /**
+   * If {@code METADATA}, search the metadata for the specified key;
+   * if {@code PARAMS}, search the params for the specified key;
+   */
+  private Keyset keyset = Keyset.METADATA;
+
+  /**
+   * The regex pattern to match in the property value (can be null to indicate
+   * that any value is considered a match).
+   */
+  private Pattern pattern;
+
+  /**
+   * If {@code found}, make a transmission decision on a match;
+   * if {@code not-found}, make a transmission decision on a failed match.
+   */
+  private When when = When.FOUND;
+
+  /**
+   * The {@code TransmissionDecision} to be made.
+   */
+  private TransmissionDecision decision;
+
+  private RegexFilter(String key, Keyset keyset, Pattern pattern,
+      When when, TransmissionDecision decision) {
+    this.key = key;
+    this.keyset = keyset;
+    this.pattern = pattern;
+    this.when = when;
+    this.decision = decision;
+  }
+
+  /**
+   * Search (only) the {@code Metadata} for an instance of the key
+   * containing a value that matches the {@code pattern}.  Returns {@code true}
+   * if found, {@code false} if not.
+   */
+  private boolean foundInMetadata(Metadata metadata) {
+    boolean found = false;
+    for (String value : metadata.getAllValues(key)) {
+      if (pattern.matcher(value).find()) {
+        found = true;
+        break;
+      }
+    }
+    log.fine((found ? "Did" : "Did not") + " find matching pattern for key `"
+        + key + "' in metadata.");
+    return found;
+  }
+
+  /**
+   * Search (only) the {@code params} for an instance of the key
+   * containing a value that matches the {@code pattern}.  Returns {@code true}
+   * if found, {@code false} if not.
+   */
+  private boolean foundInParams(Map<String, String> params) {
+    boolean found = false;
+    if (params.containsKey(key)) {
+      found = pattern.matcher(params.get(key)).find();
+    }
+    log.fine((found ? "Did" : "Did not") + " find matching pattern for key `"
+        + key + "' in params.");
+    return found;
+  }
+
+  /**
+   * Conditionally adds a {@code Transmission-Decision} entry to the
+   * {@code params Map}. The decision is based on settings of the
+   * {@code key}, {@code pattern}, {@code decideOnMatch}, {@code decision},
+   * and {@code keyset} configuration variables (as discussed above).
+   */
+  @Override
+  public void transform(Metadata metadata, Map<String, String> params) {
+    boolean found;
+    switch (keyset) {
+      case METADATA:
+        found = foundInMetadata(metadata);
+        break;
+      case PARAMS:
+        found = foundInParams(params);
+        break;
+      default:
+        found = false;	// can't happen
```

Comment:
Remove tab character. Please check CLs for lint errors generally, there are half-dozen scattered around.